// victory-line/lib/v37/victory-line.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.VictoryLineBase = exports.VictoryLine = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _curve = require("../curve");

var _victoryCore = require("victory-core");

var _clone = require("./clone");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { _defineProperty(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var propTypes = {
  interpolation: _propTypes.default.oneOfType([_propTypes.default.oneOf(["basis", "bundle", "cardinal", "catmullRom", "linear", "monotoneX", "monotoneY", "natural", "step", "stepAfter", "stepBefore"]), _propTypes.default.func]).isRequired
};
var defaultProps = {
  data: [{
    x: 1,
    y: 1
  }, {
    x: 2,
    y: 2
  }, {
    x: 3,
    y: 3
  }, {
    x: 4,
    y: 4
  }],
  containerComponent: /*#__PURE__*/_react.default.createElement(_victoryCore.VictoryContainer, null),
  dataComponent: /*#__PURE__*/_react.default.createElement(_curve.Curve, null),
  interpolation: "linear",
  labelComponent: /*#__PURE__*/_react.default.createElement(_victoryCore.VictoryLabel, {
    renderInPortal: true
  }),
  groupComponent: /*#__PURE__*/_react.default.createElement(_victoryCore.VictoryClipContainer, null),
  samples: 50,
  sortKey: "x",
  sortOrder: "ascending",
  standalone: true,
  theme: _victoryCore.VictoryTheme.grayscale
};

var VictoryLineBase = function (props) {
  var data = (0, _victoryCore.useData)();
  var scale = (0, _victoryCore.useScale)();
  var domain = (0, _victoryCore.useDomain)();

  var dataProps = _objectSpread(_objectSpread({}, props), {}, {
    data: data,
    scale: scale,
    domain: domain
  });

  return /*#__PURE__*/_react.default.createElement(_clone.Clone, {
    element: props.groupComponent
  }, /*#__PURE__*/_react.default.createElement(_clone.Clone, _extends({
    element: props.dataComponent
  }, dataProps)));
};

exports.VictoryLineBase = VictoryLineBase;
VictoryLineBase.propTypes = propTypes;
VictoryLineBase.defaultProps = defaultProps;
var VictoryLine = (0, _victoryCore.withContainer)(VictoryLineBase);
exports.VictoryLine = VictoryLine;
